src/main/scala/lms/core/codegen/c_codegen.scala

```diff
@@ -260,6 +260,14 @@ class ExtendedCCodeGen extends CompactCodeGen with ExtendedCodeGen {
   val libraryFlags = mutable.HashSet[String]()
   def registerLibrary(nLibraries: String*) = libraryFlags ++= nLibraries.toSet
 
+  val libraryPaths = mutable.HashSet[String]()
+  def registerLibraryPath(paths: String*) = libraryPaths ++= paths.toSet
```

Comment:
is the libraryPaths used anywhere in compilation? Maybe it is used but not in this repo?

Reply:
It would be used in the case where the linked library (static/dynamic) is not in the env's library paths. Not sure how often this would be encountered.

Reply:
It would be useful when we don't have permission to install libraries system-wide on some machines, under this case, we need to specify where are the .so files when compiling.

Reply:
make sense! but do we need to add something to the compilation cmd as well?

Reply:
It would be something like 
`    val libraryPaths = codegen.composePaths(codegen.libraryPaths, "-L")`
Then add it to command.